text/0017-define_dataset_creation.md

```diff
@@ -0,0 +1,220 @@
+- Feature Name: Define Dataset Creation
+- Start Date: 2018-11-15
+- RFC PR: <!-- (leave this empty) -->
+- Issue: <!-- (leave this empty) -->
+
+# Summary
+[summary]: #summary
+
+Simplify "new" and "save" into one algorithm that defines the high-level steps for creating a dataset, making learning Qri easier.
+
+# Motivation
+[motivation]: #motivations
+
+An early Qri user recently gave us some feedback:
+
+> when I was writing a transformation, it wasn't super clear what needed to happen in the sky file and what needed to happen in the dataset.yml. I started out with both and had to deal with a bunch of errors which stemmed from (I think) me accidentally specifying things in both places or in the wrong place. In particular I wasn't sure what runs first. For example, does the ds in the transformation know the schema specified in dataset.yml or does qri check the output of transform against the specified schema? 
+
+What I'm hearing is Qri doesn't have clear mental model of what will happen when the user runs `qri new` or  `qri save`. In some ways this makes sense, Qri is brand new. For that same reason it's important to introduce a consistent, digestable mental model as early as possible.
```

Comment:
The feedback also points to a possible ambiguity between transform and dataset.yml, since both can do similar things. This is a distinct (but admittedly related) problem from the semantics of `qri save`. I understand it's hard to answer the original question without cleaning up our mental model, but I think it's also important to address the original question directly.

Reply:
agreed. I've added some language that better connects the RFC content to Matt's feedback

Reply:
Very good addition, but the next paragraph doesn't really start as well as it could, now that it doesn't immediately follow the "feedback" text. Maybe start it more like "But in order to properly respond to feedback such as this...."